base/commandLine.cpp

```diff
@@ -1039,8 +1175,15 @@ bool processSettings(Common::String &command, Common::StringMap &settings, Commo
 		return true;
 	} else if (command == "auto-detect") {
 		// If auto-detects succeed, we want to return false so that the game is started
-		return !autoDetect(settings["path"]);
+		return !detectGames(settings["path"], true);
 		//return true;
+	} else if (command == "detect") {
+		detectGames(settings["path"], false);
+		return true;
```

Comment:
This function is called in the main function in main.cpp.
`	// TODO: deal with settings that require plugins to be loaded
	if (Base::processSettings(command, settings, res)) {
		if (res.getCode() != Common::kNoError)
			warning("%s", res.getDesc().c_str());
		return res.getCode();
	}
       // open scummvm interface/run game instructions...`
I guess the return value of this function means: true->do nothing except for displaying informations in command line, false-> open scummvm gui/run game/... next
So I choose to always return true for "detect" means, I only want the game informations displayed and nothing else whatever the detect result is. But it"s free to change this return value depending on what we need.